src/reflect/scala/reflect/internal/pickling/UnPickler.scala

```diff
@@ -224,10 +224,9 @@ abstract class UnPickler {
           if (owner.isOverloaded)
             return NoSymbol
 
-          if (tag == EXTMODCLASSref) {
+          if (tag == EXTMODCLASSref && !owner.isInstanceOf[StubSymbol])
             owner.info.decl(nme.moduleVarName(name.toTermName))
-          }
-          NoSymbol
+          else NoSymbol
```

Comment:
Oh right, I didn't even see that. Guess I was blinded by the early return above!

It's been a no-op since https://github.com/scala/scala/commit/df3689139c4d4bcd2933364d13b8195c3433eb43#diff-7bf0456105421b3217bd1d63b8c864d4L229, the original code was added in 020053c3215579e8aeb871a4ad0078516994270d.

I'll poke at it and see if there is a bug. Otherwise we can drop this tactic from unpickler altogether.